### ISSUE TITLE
Incremental Save following a load

### ISSUE THREAD

**Issue by acurrieclark** [automerge/automerge#592]
**TLDR:** Following loading a document from incremental saves, should the next call to `saveIncremental` be based on the loaded state?

---
Automerge Repo makes use of incremental loads when storing documents locally. The current process is:

1. Save a snapshot of the whole document
2. save incremental changes up to a certain limit
3. when the limit is reached, clear the incremental saves and return to step 1

For example, a saved document state might be:

```
{doc-id}.snapshot -> initial save
{doc-id}.increment.1 -> first incremental save
{doc-id}.increment.2 -> second incremental save
```

The issue arrives if we load all the increments using `loadIncremental` and then attempt to save incrementally again. For example, the next change in the above example would create:

```
{doc-id}.snapshot -> initial save (whole doc)
{doc-id}.increment.1 -> first incremental save
{doc-id}.increment.2 -> second incremental save
{doc-id}.increment.3 -> *** this is a save of the whole doc again ***
```

In Automerge prior to `2.0.3`, trying to load the above state would cause a `mismatched heads` error. While the error is no longer thrown, it is clearly not ideal to be bloating the local store.

This is really a question of expectation. If you load a document using `incrementalLoad`, should the next `saveIncremental` be anchored to the load, or would you expect it to save the whole doc?

The issue can be seemingly worked around by calling `saveIncremental(Doc)`, but perhaps the alternative should be the default?

**Comment by alexjg** [comment]
I think the logic of `saveincremental` is that it should save everything since the last time you called it. I don't really love this API because there's now a piece of state hidden in the document which you can't really reason about. In other bindings I've instead used a `saveIncrementalSince(heads)` API, which is more work because now you have to hold on to some heads, but also much clearer because _you're_ holding onto those heads, not the internals of the document. Maybe we should add such an API here?

**Comment by acurrieclark** [comment]
That would be ideal. Perhaps as an optional parameter for now, which could be made required in a future release?
